pkg/deploy/generator/resources_rp.go

```diff
@@ -1079,28 +1099,18 @@ func (g *generator) database(databaseName string, addDependsOn bool) []*arm.Reso
 		},
 	}
 
-	if g.production {
-		portal.Resource.(*mgmtdocumentdb.SQLContainerCreateUpdateParameters).SQLContainerCreateUpdateProperties.Options.Throughput = to.Int32Ptr(400)
-		gateway.Resource.(*mgmtdocumentdb.SQLContainerCreateUpdateParameters).SQLContainerCreateUpdateProperties.Options.Throughput = to.Int32Ptr(400)
+	if !g.production {
+		database.Resource.(*mgmtdocumentdb.SQLDatabaseCreateUpdateParameters).SQLDatabaseCreateUpdateProperties.Options = &mgmtdocumentdb.CreateUpdateOptions{
+			AutoscaleSettings: &mgmtdocumentdb.AutoscaleSettings{
+				MaxThroughput: to.Int32Ptr(1000),
+			},
+		}
+		portal.Resource.(*mgmtdocumentdb.SQLContainerCreateUpdateParameters).SQLContainerCreateUpdateProperties.Options = &mgmtdocumentdb.CreateUpdateOptions{}
+		gateway.Resource.(*mgmtdocumentdb.SQLContainerCreateUpdateParameters).SQLContainerCreateUpdateProperties.Options = &mgmtdocumentdb.CreateUpdateOptions{}
```

Comment:
Can you comment on why we need this, and only if not prod? Is that to remove any configuration we've already set above for throughput? So, for dev, we expect no configuration to exist, and we just throw in auto-scaling instead?

Reply:
Yes, the intended state is:

Development: Autoscaling throughput enabled on top-level database, no container-specific throughputs defined
Production: Static throughput enabled on top-level database, as well as container-specific throughputs on portal/gateway containers.